devicetypes/axis/axis-gear-st.src/axis-gear-st.groovy

```diff
@@ -0,0 +1,408 @@
+	if (state.currentVersion >= MIN_WINDOW_COVERING_VERSION){
+		zigbee.command(CLUSTER_WINDOWCOVERING, WINDOWCOVERING_CMD_CLOSE)
+	}
+	else {
+		setLevel(0)
+	}
+}
+
+def presetPosition() {
+	log.info "presetPosition()"
+	setLevel(preset ?: state.preset ?: 50)
+}
+
+//Reporting of Battery & position levels
+def ping(){
+	log.debug "Ping() "
+	return refresh()
+}
+
+//Set blind State based on position (which shows appropriate image) 
+def setWindowShade(value) {
+	if ((value>0)&&(value<99)){
+		sendEvent(name: "windowShade", value: "partially open", displayed:true)
+	}
+	else if (value >= 99) {
+		sendEvent(name: "windowShade", value: "open", displayed:true)
+	}
+	else {
+		sendEvent(name: "windowShade", value: "closed", displayed:true)
+	}
+}
+
+//Refresh command
+def refresh() {
+	log.debug "parse() refresh"
+	def cmds_refresh = null
+    
+	if (state.currentVersion >= MIN_WINDOW_COVERING_VERSION){
+		cmds_refresh = zigbee.readAttribute(CLUSTER_WINDOWCOVERING, WINDOWCOVERING_ATTR_LIFTPERCENTAGE) +
+							zigbee.readAttribute(CLUSTER_POWER, POWER_ATTR_BATTERY) +
+							zigbee.readAttribute(CLUSTER_BASIC, BASIC_ATTR_SWBUILDID)
+    }
+    else {
+		cmds_refresh = zigbee.readAttribute(CLUSTER_LEVEL, LEVEL_ATTR_LEVEL) +
+							zigbee.readAttribute(CLUSTER_POWER, POWER_ATTR_BATTERY) +
+							zigbee.readAttribute(CLUSTER_BASIC, BASIC_ATTR_SWBUILDID)
+	}
+	log.info "refresh() --- cmds: $cmds_refresh"
+	
+	return cmds_refresh
+}
+
+def getversion () {
+	//state.currentVersion = 0
+	sendEvent(name: "version", value: "Checking Version ... ")     
+	return zigbee.readAttribute(CLUSTER_BASIC, BASIC_ATTR_SWBUILDID)
+}
+
+//configure reporting
+def configure() {   
+	state.currentVersion = 0
+	sendEvent(name: "windowShade", value: "unknown")
+	log.debug "Configuring Reporting and Bindings."
+	sendEvent(name: "checkInterval", value: (2 * 60 * 60 + 10 * 60), displayed: true, data: [protocol: "zigbee", hubHardwareId: device.hub.hardwareID])
+
+	def attrs_refresh = zigbee.readAttribute(CLUSTER_BASIC, BASIC_ATTR_SWBUILDID) +
+						zigbee.readAttribute(CLUSTER_WINDOWCOVERING, WINDOWCOVERING_ATTR_LIFTPERCENTAGE) +
+						zigbee.readAttribute(CLUSTER_ONOFF, ONOFF_ATTR_ONOFFSTATE) +
+						zigbee.readAttribute(CLUSTER_LEVEL, LEVEL_ATTR_LEVEL) +
+						zigbee.readAttribute(CLUSTER_POWER, POWER_ATTR_BATTERY)
+						
+    def cmds = zigbee.configureReporting(CLUSTER_WINDOWCOVERING, WINDOWCOVERING_ATTR_LIFTPERCENTAGE, 0x20, 1, 3600, 0x00) + 
+				zigbee.configureReporting(CLUSTER_ONOFF, ONOFF_ATTR_ONOFFSTATE, 0x20, 1, 3600, 0x00) +
+				zigbee.configureReporting(CLUSTER_LEVEL, LEVEL_ATTR_LEVEL, 0x20, 1, 3600, 0x00) +
+				zigbee.configureReporting(CLUSTER_POWER, POWER_ATTR_BATTERY, 0x20, 1, 3600, 0x01)
+				
+	log.info "configure() --- cmds: $cmds"
+	return attrs_refresh + cmds
+}
+
+def parse(String description) {
+	log.trace "parse() --- description: $description"
+	
+	def event = zigbee.getEvent(description)
+	if (event) {
+		sendEvent(event)
```

Comment:
I believe this will result in you sending duplicate `level` events. What might be a better alternative is to do some inspections of level events returned by this function here and then generate the supplementary windowShade events and then only conditionally call `parseReportAttributeMessage` if no event was returned.

Reply:
change made to only process the onoff events to correctly report when used via device operation (off indicates closed position). All other attribute reports are handled in parseReportAttributeMessage